tests/NuGet.Services.AzureSearch.Tests/Auxiliary2AzureSearch/DownloadSetComparerFacts.cs

```diff
@@ -0,0 +1,192 @@
+                var newData = new DownloadData();
+                newData.SetDownloadCount(IdA, V1, 5); // No change
+                newData.SetDownloadCount(IdA, V2, 2); // Added
+
+                var delta = Target.Compare(oldData, newData);
+
+                Assert.Equal(KeyValuePair.Create(IdA, 7L), Assert.Single(delta));
+                VerifyDecreaseTelemetry(Times.Never());
+            }
+
+            [Fact]
+            public void DetectsIncreaseDueToAddedId()
+            {
+                var oldData = new DownloadData();
+                var newData = new DownloadData();
+                newData.SetDownloadCount(IdA, V1, 5); // Added
+                newData.SetDownloadCount(IdA, V2, 2); // Added
+
+                var delta = Target.Compare(oldData, newData);
+
+                Assert.Equal(KeyValuePair.Create(IdA, 7L), Assert.Single(delta));
+                VerifyDecreaseTelemetry(Times.Never());
+            }
+
+            [Fact]
+            public void DetectsDecreaseDueToRealDecrease()
+            {
+                var oldData = new DownloadData();
+                oldData.SetDownloadCount(IdA, V1, 7);
+                oldData.SetDownloadCount(IdA, V2, 1);
+                var newData = new DownloadData();
+                newData.SetDownloadCount(IdA, V1, 5);
+                newData.SetDownloadCount(IdA, V2, 1);
+
+                var delta = Target.Compare(oldData, newData);
+
+                Assert.Equal(KeyValuePair.Create(IdA, 6L), Assert.Single(delta));
+                VerifyDecreaseTelemetry(Times.Once());
+                TelemetryService.Verify(
+                    x => x.TrackDownloadCountDecrease(
+                        IdA,
+                        V1,
+                        true,
+                        true,
+                        7,
+                        true,
+                        true,
+                        5),
+                    Times.Once);
+            }
+
+            [Fact]
+            public void DetectsDecreaseDueToMissingVersion()
+            {
+                var oldData = new DownloadData();
+                oldData.SetDownloadCount(IdA, V1, 7);
+                oldData.SetDownloadCount(IdA, V2, 1);
+                var newData = new DownloadData();
+                newData.SetDownloadCount(IdA, V2, 1);
+
+                var delta = Target.Compare(oldData, newData);
+
+                Assert.Equal(KeyValuePair.Create(IdA, 1L), Assert.Single(delta));
+                VerifyDecreaseTelemetry(Times.Once());
+                TelemetryService.Verify(
+                    x => x.TrackDownloadCountDecrease(
+                        IdA,
+                        V1,
+                        true,
+                        true,
+                        7,
+                        true,
+                        false,
+                        0),
+                    Times.Once);
+            }
+
+            [Fact]
+            public void DetectsDecreaseDueToMissingId()
```

Comment:
Should you test the case where V1 increases by X, but V2 decreases by X, resulting in an empty delta but decrease telemetry? It's an weird/obscure corner case, feel free to ignore

Reply:
No this is a good thing to test. I'll add that.